docs/sources/mimir/manage/mimir-runbooks/_index.md

```diff
@@ -2485,6 +2485,91 @@ gsutil cp $file ${file%#*}
+
+Mimir publishes "distroless" container images. A [distroless image](https://github.com/GoogleContainerTools/distroless/blob/main/README.md)
+contains very little outside of what is needed to run a single binary.
+They don't tend to include any text editors, process managers, package managers, or other debugging tools.
+
+This can pose a challenge when diagnosing potential problems. There exists no shell inside the container
+to attach to or any tools to inspect configuration files and so on.
+
+However, for debugging distroless containers we can take the approach of attaching a more complete
+container to the existing containers namespace. This allows us to bring in all of the
+tools we may need and to not disturb the existing environment.
+That is, we do not need to restart the running container to attach our debug tools.
+
+## Creating a debug container
+
+Kubernetes gives us a command that allows us to start an ephemeral debug container in a pre-existing pod
+attaching it to the same namespace as other containers in that pod. More detail can be read about the command and
+[how to debug running pods in the Kubernetes docs](https://kubernetes.io/docs/tasks/debug/debug-application/debug-running-pod/#ephemeral-container).
+
+```bash
+kubectl --namespace mimir debug -it pod/compactor-0 --image=ubuntu:latest --target=compactor -c mimir-debug-container
+```
+
+- `pod/name` is the pod to attach to.
+- `--target=` is the container within that pod with which to share a kernel namespace.
+- `--image=` is the image of the debug container you wish to use.
+- `-c` is the name to use for the ephemeral container. This is optional, but useful if you want to re-use it.
+
+You can now see all of the processes running in this space. For example:
+
+```
+/ # ps aux
+PID   USER     TIME  COMMAND
+    1 root      5:36 /usr/bin/mimir -flags
+   31 root      0:00 /bin/bash
+   36 root      0:00 ps aux
+```
+
+The PID 1 is the process that is executed in the target container. You can now use
+tools within your debug image to interact with the running process. However, note
+that your root path and important environment variables like $PATH will be different to
+that of the target container.
+
+To access the root filesystem of the target container, you can find it in `/proc/1/root`. For
+example, the commonly used data directory would be found at `/proc/1/root/data`, and
+binaries of the target container would be somewhere like `/proc/1/root/usr/bin/mimir`.
+
+## Copying files from a distroless container
+
+Because distroless images do not have `tar` in them, it is not possible to copy files using `kubectl cp`.
+
+To work around this, we can create a debug container attached to the pod (as per above) and then use `kubectl cp` against that.
+The debug container cannot have terminated in order for us to be able to use it. This means if you run a debug container to get a shell,
+you need to keep the shell open in order to do the following.
```

Comment:
[nit]

```suggestion
The debug container must remain running while copying files. This means if you run a debug container to get a shell,
you need to keep the shell open in order to do the following.
```

Reply:
Terminated is a state the k8s uses, so I think we should keep this.